solidity/test/bridge/Bridge.test.ts

```diff
@@ -4669,6 +4789,961 @@ describe("Bridge", () => {
+      })
+
+      it("should revert", async () => {
+        await expect(
+          bridge
+            .connect(thirdParty)
+            .submitFraudChallenge(
+              fraudWalletPublicKey,
+              data.sighash,
+              incorrectV,
+              data.signature.r,
+              data.signature.s,
+              {
+                value: fraudChallengeDepositAmount,
+              }
+            )
+        ).to.be.revertedWith("Signature verification failure")
+      })
+    })
+
+    context("when incorrect signature data is used", () => {
+      // just swap r and s
+      const incorrectS = data.signature.r
+      const incorrectR = data.signature.s
+
+      before(async () => {
+        await createSnapshot()
+        await bridge.setWallet(fraudWalletPublicKeyHash, {
+          state: 1,
+          pendingRedemptionsValue: 0,
+        })
+        await bridge
+          .connect(governance)
+          .setFraudChallengeDepositAmount(fraudChallengeDepositAmount)
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+
+      it("should revert", async () => {
+        await expect(
+          bridge
+            .connect(thirdParty)
+            .submitFraudChallenge(
+              fraudWalletPublicKey,
+              data.sighash,
+              data.signature.v,
+              incorrectR,
+              incorrectS,
+              {
+                value: fraudChallengeDepositAmount,
+              }
+            )
+        ).to.be.revertedWith("Signature verification failure")
+      })
+    })
+
+    context("when the same fraud challenge called twice", () => {
+      before(async () => {
+        await createSnapshot()
+        await bridge
+          .connect(governance)
+          .setFraudChallengeDepositAmount(fraudChallengeDepositAmount)
+        await bridge.setWallet(fraudWalletPublicKeyHash, {
+          state: 1,
+          pendingRedemptionsValue: 0,
+        })
+        await bridge
+          .connect(thirdParty)
+          .submitFraudChallenge(
+            fraudWalletPublicKey,
+            data.sighash,
+            data.signature.v,
+            data.signature.r,
+            data.signature.s,
+            {
+              value: fraudChallengeDepositAmount,
+            }
+          )
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+
+      it("should revert", async () => {
+        await expect(
+          bridge
+            .connect(thirdParty)
+            .submitFraudChallenge(
+              fraudWalletPublicKey,
+              data.sighash,
+              data.signature.v,
+              data.signature.r,
+              data.signature.s,
+              {
+                value: fraudChallengeDepositAmount,
+              }
+            )
+        ).to.be.revertedWith("Fraud challenge already exists")
+      })
+    })
+
+    context(
+      "when the provided data is correct and amount of ether sent is enough",
```

Comment:
Can we put the happy-path case at the top of the context? That helps to understand the expectations.

Reply:
Done in 61efd70c5ee84fcc4ac10154b2d2cd8c9af0a5aa.
I structured unit tests for all added functions in a similar manner as in redemption tests.